src/main/frontend/src/app/app.service.ts

```diff
@@ -15,6 +15,14 @@ export type GitHubState = (typeof GitHubStates)[keyof typeof GitHubStates];
 export class AppService {
   private http = inject(HttpClient);
 
+  /**
+   * Fetches the CSRF token from the backend to ensure the XSRF-TOKEN cookie is set.
+   * This should be called on app initialization.
+   */
+  public initializeCsrfToken(): Observable<void> {
+    return this.http.get<void>(this.createAPIUrl('csrf'));
+  }
```

Comment:
What is the purpose of this?
CSRF/XSRF tokens are by default send through a cookie to the client, the client sends it back to the server through a header. Both Spring and Angular support this as default behaviour, so I dont get what you're doing with this request